### PR TITLE
Add unittest for  flatten2_matmul squeeze2_matmul reshape2_matmul pass

### PR DESCRIPTION
**Pull request by heliqi**
### PR types
Bug fixes

### PR changes
Others

### Describe
Add unittest for flatten2_matmul_fuse_pass squeeze2_matmul_fuse_pass reshape2_matmul_fuse_pass:
1.   Add a test code for squeeze2_matmul_fuse_pass , fix "squeeze2" name error in OpCompat input.
2.  Add a test code for flatten2_matmul_fuse_pass ,     remove the false attribute "Y" on the op "flatten2" and fix "axis" attribute  error limits.
3.  Add a test code for reshape2_matmul_fuse_pass,  and fix squeeze2's name of OpCompat input
4.  限制flatten的axis必须为1， 否则 flaten + mamul 被fuse成 mul后， 输出形状对不上。 
     比如:   x[a, b, c, d] -> flatten(axis = 2) -> [a*b, c*d] -> matmul(y=[c*d, e]) -> output: [a*b, e]
                x [a, b, c, d]  y[c*d, e]-> mul(x_num_axis = 2) -> [a, b, e]

### CODE REVIEW

File: paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc [hunk 1]
```diff
@@ -169,17 +169,14 @@ Flatten2MatmulFusePass::Flatten2MatmulFusePass() {
...
-      .End()
       .AddOutput("Out")
       .IsTensor()
       .End()
       .AddOutput("XShape")
       .IsTensor()
       .End()
       .AddAttr("axis")
-      .IsNumGE(0)
+      .IsNumEQ(1)
```
Comment (jiangjiajun) [inline paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:179]: 为什么需要限制为1，在PR描述中说明下原因

Comment (heliqi) [reply paddle/fluid/framework/ir/map_matmul_to_mul_pass.cc:179]: 否则输出形状对不上， 已在PR中加上

**Review by jiangjiajun — DISMISSED** [review]
LGTM

**Review by XieYunshen — DISMISSED** [review]
LGTM for 
`  set_tests_properties(test_flatten2_matmul_fuse_pass PROPERTIES TIMEOUT 240)
  set_tests_properties(test_squeeze2_matmul_fuse_pass PROPERTIES TIMEOUT 240)
  set_tests_properties(test_reshape2_matmul_fuse_pass PROPERTIES TIMEOUT 240)`

**Review by XieYunshen — DISMISSED** [review]

File: python/paddle/fluid/tests/unittests/ir/inference/test_flatten2_matmul_fuse_pass.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        else:
+            bias_shape = [flatten_shape[0], y_shape[1]]
+            if draw(st.booleans()):
+                bias_shape[1] = 1
+
+        flatten2_op = OpConfig(
+            "flatten2",
+            inputs={
+                "X": ["flatten2_x"],
+                # "Y": ["flatten2_y"]
```
Comment (jiangjiajun) [inline python/paddle/fluid/tests/unittests/ir/inference/test_flatten2_matmul_fuse_pass.py]: 删除

File: python/paddle/fluid/tests/unittests/ir/inference/test_flatten2_matmul_fuse_pass.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                    # "flatten2_y": TensorConfig(shape=x_shape),
+                },
+                outputs=ops[-1].outputs["Out"], )
+        else:
+            program_config = ProgramConfig(
+                ops=ops,
+                weights={},
+                inputs={
+                    "flatten2_x": TensorConfig(shape=x_shape),
+                    # "flatten2_y": TensorConfig(shape=x_shape),
```
Comment (jiangjiajun) [inline python/paddle/fluid/tests/unittests/ir/inference/test_flatten2_matmul_fuse_pass.py]: 删除

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

**Review by jiangjiajun — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]

**Review by jiangjiajun — APPROVED** [review]

**Pull request merged**